File: python/medium/a_circle_and_two_squares.py
# A Circle and Two Squares
# https://edabit.com/challenge/NNhkGocuPMcryW7GP

import math

def square_areas_difference(r):
  square_root = math.sqrt(2)
  small_hypo = 2*r
  small_side = small_hypo/square_root
  small_area = small_side**2

  big_side = small_hypo
  big_area = big_side**2

  return int(big_area-small_area)

File: javascript/medium/a_circle_and_two_squares.js
// A Circle and Two Square
// https://edabit.com/challenge/ApTfL3DXJtyAmiajv

// side of big square = radius * 2
// hypotenuse of small circle = radius * 2
// a2 + b2 = radius * 2
// a = b = csqrt2

function squareAreasDifference(r) {
    var sqrt_2 = Math.sqrt(2);
    var diameter = 2 * r;
    var small_side = diameter / sqrt_2;

    var big_square = diameter * diameter;
    var small_square = small_side * small_side;

    return Math.round(big_square - small_square);
}